.github/files/coverage-munger/upload-coverage.sh

```diff
@@ -95,3 +101,82 @@ done
 do_req "op=finish&token=$TOKEN"
 TOKEN=
 echo '::endgroup::'
+
+if [[ "$PR_ID" != "trunk" ]]; then
+	echo "::group::Setting GitHub status"
+	if jq -e '.covinfo' <<<"$JSON" &>/dev/null; then
+		JSON=$( jq '.covinfo' <<<"$JSON" )
+		if [[ "$STATUS" != 'success' ]]; then
+			JSON=$( jq '.state |= "pending" | .description |= "Waiting for tests to pass" | .msg |= "Cannot generate coverage summary while tests are failing. :zipper_mouth_face:\n\nPlease fix the tests, or re-run the Code coverage job if it was something being flaky."' <<<"$JSON" )
+		fi
+	else
+		JSON='{"state":"error","description":"No covinfo received from server","msg":"","footer":""}'
+	fi
+	jq . <<<"$JSON"
+	curl -v -L --fail \
+		--url "${GITHUB_API_URL}/repos/${GITHUB_REPOSITORY}/statuses/$( jq --arg V "$PR_HEAD" -nr '$V | @uri' )" \
+		--header "authorization: Bearer $API_TOKEN_GITHUB" \
+		--header 'content-type: application/json' \
+		--data "$( jq -c --arg PR "$PR_ID" '{
+			context: "Code coverage requirement",
+			state: .state,
+			target_url: "https://jetpackcodecoverage.atomicsites.blog/prs/\( $PR | @uri )/",
+			description: .description,
+		}' <<<"$JSON" )"
+	echo "::endgroup::"
+
+	# Find the last comment starting with "### Code Coverage Summary"
+	echo "::group::Looking for existing comment"
+	PAGE=1
+	while true; do
+		J=$( curl -v -L fail \
```

Comment:
J is for Jorsch, right?

Reply:
Actually it's for "I already used `JSON` and was too lazy to type `JSON2` or think of something better" 😀